winml/lib/Api.Core/WinMLAdapter.cpp

```diff
@@ -630,68 +629,80 @@ class WinMLAdapter : public Microsoft::WRL::RuntimeClass<
 #ifdef USE_DML
     auto impl = wil::MakeOrThrow<AbiCustomRegistryImpl>();
     *registry = impl.Detach();
-  return S_OK;
+    return S_OK;
 #else
-  return E_NOTIMPL;
+    return E_NOTIMPL;
 #endif USE_DML
-}
+  }
 
-void* STDMETHODCALLTYPE CreateGPUAllocationFromD3DResource(ID3D12Resource* pResource) override {
+  HRESULT STDMETHODCALLTYPE GetOperatorRegistry(ILearningModelOperatorProviderNative* operator_provider_native, IMLOperatorRegistry** registry) override {
 #ifdef USE_DML
-  return Dml::CreateGPUAllocationFromD3DResource(pResource);
+    // Retrieve the "operator abi" registry.
+    winrt::com_ptr<IMLOperatorRegistry> operator_registry;
+    operator_provider_native->GetRegistry(operator_registry.put());
```

Comment:
>GetRegistry [](start = 30, length = 11)

Woudn't GetRegistry return an HRESULT? S_OK would always be returned from this function if this isn't handled.